plugins/plugin-webpack/test/serializer.js

```diff
@@ -0,0 +1,71 @@
+module.exports = {
+  getSnowpackPluginOutputSnapshotSerializer,
+};
+
+const {format} = require('util');
+
+const strpAnsi = require('strip-ansi');
+
+/**
+ * Serializer of written files as well as console.log output.
+ *
+ * Both file contents and log output is normalized to account for differences
+ * in UNIX and Windows systems.
+ *
+ * @param string basePath all files written outside this path will be ignored (usually set to __dirname)
+ */
+function getSnowpackPluginOutputSnapshotSerializer(basePath) {
+  return {
+    serialize(value) {
+      if (value.getMockName() === 'console.log') {
+        return value.mock.calls
+          .map(toSingleArgument)
+          .map(toNoralizedByteSize)
+          .map(removeColors)
+          .join('\n');
+      }
+
+      const calls = value.mock.calls
+        .filter(isLocal)
+        .map(toPathAndStringContent.bind(null, basePath));
+      return calls
+        .sort((a, b) => {
+          return a[0] < b[0] ? -1 : 1;
+        })
+        .map(([path, content]) => {
+          return `FILE: ${path}\n${content}`;
+        })
+        .join(
+          '\n\n--------------------------------------------------------------------------------\n\n',
+        );
+    },
+
+    test(value) {
+      return value && value.mock;
+    },
+  };
+}
+
+function toPathAndStringContent(basePath, [path, content]) {
+  const shortPath = path.replace(basePath, '').substr(1);
+  // unix-ify folder separators for Windows
+  const normalizedPath = shortPath.replace(/\\/g, '/');
```

Comment:
Could this possibly be replacing a backslash out-of-context when it doesn’t need to? I think it’s probably fine; just wanted to raise that.

Reply:
I don't think so, the replace is only run on the `path` argument of `fs.writeFile`/`fs.writeFileSync`, not the content or log output

Reply:
Ah OK. Yeah I ask because we had a problem with Rollup where it inlined comments with things like `/* src\App.js */` in the actual code, so we had to find a way to replace those on Windows (I think we just ended up stripping out comments for comparison). Moral of the story is you never know where backslashes will pop up! 🙃